sparse/core.py

```diff
@@ -283,7 +283,10 @@ def __getitem__(self, index):
             coords.append(self.coords[j][mask])
             shape.append(self.shape[j])
 
-        coords = np.stack(coords, axis=0)
+        if coords:
+            coords = np.stack(coords, axis=0)
+        else:
+            coords = np.array(coords)
```

Comment:
I may be missing something here, but won't `np.array([])` create an array with a `dtype` of `float64` and a shape of `(0,)`? It would be much nicer to do this: `coords = np.empty((0, len(shape)), dtype=np.uint8)`

Reply:
And also to specify a `dtype` even in the first case...